setup-env/action.yaml

```diff
@@ -20,14 +20,48 @@ runs:
           echo "package_manager=npm" >> $GITHUB_OUTPUT
           echo "install_cmd=npm ci" >> $GITHUB_OUTPUT
           echo "hash_glob=**/package-lock.json" >> $GITHUB_OUTPUT
+          FOUND_NODE=true
         elif [ -e yarn.lock ]; then
           echo "package_manager=yarn" >> $GITHUB_OUTPUT
           echo "install_cmd=yarn install --immutable" >> $GITHUB_OUTPUT
           echo "hash_glob=**/yarn.lock" >> $GITHUB_OUTPUT
+          FOUND_NODE=true
         elif [ -e pnpm-lock.yaml ]; then
           echo "package_manager=pnpm" >> $GITHUB_OUTPUT
           echo "install_cmd=pnpm install --frozen-lockfile" >> $GITHUB_OUTPUT
           echo "hash_glob=**/pnpm-lock.yaml" >> $GITHUB_OUTPUT
+          FOUND_NODE=true
+        fi
+
+        if [ -v $FOUND_NODE ]; then
+          exit 0
+        fi
+
+        if [ -e .nvmrc ]; then
+          echo "node_version_file=.nvmrc" >> $GITHUB_OUTPUT
+          FOUND_NODE_VERSION=1
+        elif [ -e .node-version ]; then
+          echo "node_version_file=.node-version" >> $GITHUB_OUTPUT
+          FOUND_NODE_VERSION=1
+        elif [ -e .tool-versions ]; then
+          echo "node_version_file=.tool-versions" >> $GITHUB_OUTPUT
+          FOUND_NODE_VERSION=1
+        elif [ -e package.json ]; then
+          echo "node_version_file=package.json" >> $GITHUB_OUTPUT
+          FOUND_NODE_VERSION=1
+        fi
+
+        if [ -v $FOUND_NODE_VERSION ]; then
+          exit 0
+        fi
+
+        echo "Found node lockfiles but no node version!"
+
+        if which node > /dev/null; then
+          echo "Detected existing node install"
+          echo "run_install_node_packages=true" >> $GITHUB_OUTPUT
+        else
+          echo "Could not find existing node install, skipping node package installation"
         fi
```

Comment:
if this is an alternative to finding a 'node version file' - it needs to come before the `exit`, right?

Reply:
I cleaned up the logic and removed the environment variables. The logic for the first exit was inverted. The logic should be:
```
if we don't find a lockfile:
    quit - no need to do node things
if we find any node version file:
    quit and use that file in setup_node

if we find an existing node install:
    don't setup_node, but still install packages later
else:
    don't setup_node or install packages
```